crypto/compactcert/builder.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package compactcert
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/crypto/merklearray"
+	"github.com/algorand/go-algorand/data/basics"
+)
+
+//msgp:ignore sigslot
+type sigslot struct {
+	// part is the participant signing this message.  The participant
+	// information is tracked here for convenience, but it does not
+	// appear in the commitment to the sigs array.
+	part Participant
```

Comment:
I think having part in sigslot is confusing. The only thing you use for it is part.Weight (lines 75, 156, and 177), which is either 0 or participants[relevant index].Weight). Why not just have sigslot.Weight, initlized to 0, and replace line 104 with b.sigs[pos].Weight = p.Weight?

Reply:
No reason.  Storing just the weight indeed seems cleaner; changed to that.